src/main/java/me/totalfreedom/totalfreedommod/CommandSpy.java

```diff
@@ -41,4 +44,28 @@ public void onPlayerCommandPreprocess(PlayerCommandPreprocessEvent event)
         }
     }
 
+    // This will check if the user joining has command spy enabled, and if they are an admin. If they are not an admin it will ensure their command spy access has been removed, and if they are it will check to see if the admin has left command spy enabled or not on previous use.
+    @EventHandler(priority = EventPriority.MONITOR)
+    public void onPlayerJoin(PlayerJoinEvent event)
+    {
+        Admin admin = getAdmin((Player) event.getPlayer());
+        FPlayer playerdata = plugin.pl.getPlayer(event.getPlayer());
+
+        if (plugin.al.isAdmin(event.getPlayer()))
```

Comment:
Given I'm struggling to understand what that even does... I will leave the PR as is for now, its already broken and needs more work, but I can't see what I've screwed up.